File: C++Projects/POO_3964_Clase03_HerenciaMultipleYPackage/Animals/Gato.cpp
//
// Created by Diego Cruz on 19/03/2024.
//

#include "Gato.h"

Gato::Gato(string nombre) : Mamifero(nombre) {}

string Gato::expresarse() {
    return "Miau";
}

string Gato::recorridoEnMetrosPorSegundo() {
    return "¡Avanzo 4 metros en un segundo!";
}

// Getter y Setter

int Gato::getEdad() {
    return edad;
}
void Gato::setEdad(int edad) {
    this->edad = edad * 7;
}

File: JavaProjects/POO_3964_Clase04_HerenciaMultiple/src/Animals/Gato.java
package Animals;

import ParentClasses.Mamifero;
import ParentClasses.Animal;

public class Gato extends Mamifero implements Animal {
    // Constructor
    public Gato(String nombre){
        super(nombre);
    }

    @Override
    public void expresarse() {
        System.out.println("Miau");
    }

    @Override
    public String recorridoEnMetrosPorSegundo() {
        return "¡Corro 4 metros en un segundo!";
    }

    @Override
    public void alimentarse(String tipo) {
        if (tipo.equals("hogareño")){
            System.out.println("Vivo en un hogar, me alimento de concentrado");
        }
        else if (tipo.equals("callejero")){
            System.out.println("Me alimento de lo que encuentro en la calle");
        }
        else {
            System.out.println("No han definido de qué me alimento");
        }
    }
}
